File: 20190826-java-dependency-breaking-katas/src/main/java/org/codecop/dependencies/b/MarketingCampaign.java
package org.codecop.dependencies.b;

import java.time.DayOfWeek;
import java.time.LocalDateTime;

public class MarketingCampaign {

    public boolean isActive() {
        return milliSeconds() % 2 == 0;
    }

    protected long milliSeconds() {
        return System.currentTimeMillis();
    }

    public boolean isCrazySalesDay() {
        return dayOfWeek().compareTo(DayOfWeek.FRIDAY) == 0;
    }

    protected DayOfWeek dayOfWeek() {
        return LocalDateTime.now().getDayOfWeek();
    }
}

File: 20190915-csharp-dependency-breaking-katas/DependencyBreakingKatas/B/MarketingCampaign.cs
using System;

namespace Dependencies.B
{
    public class MarketingCampaign
    {
        public bool IsActive()
        {
            return MilliSeconds() % 2 == 0;
        }

        protected virtual long MilliSeconds()
        {
            return (long) DateTime.Now.TimeOfDay.TotalMilliseconds;
        }

        public bool IsCrazySalesDay()
        {
            return DayOfTheWeek().Equals(DayOfWeek.Friday);
        }

        protected virtual DayOfWeek DayOfTheWeek()
        {
            return DateTime.Now.DayOfWeek;
        }
    }
}